racket/collects/raco/raco.rkt

```diff
@@ -38,13 +39,48 @@
   ((executable-yield-handler) result) ; to enable GUI commands
   (exit result))
 
+(define (parse-cmd-output-option cmdline)
+  (let* ([tout-idx (index-of cmdline "--cmd-output")]
+         [cmd-is-not-help (not (and (pair? cmdline)
+                                    (equal? "help" (car cmdline))))])
+    (if (and tout-idx
+             (< tout-idx (sub1 (length cmdline)))
+             cmd-is-not-help)
+        (string->path (list-ref cmdline (add1 tout-idx)))
+        #f)))
+
+(define (remove-cmd-output-option cmdline cmd-output-file)
+  (if (not (path? cmd-output-file))
+      cmdline
+      (let ([tout-idx (index-of cmdline "--cmd-output")]
+            [tout-file-idx (index-of cmdline (path->string cmd-output-file))])
+        (append (take cmdline tout-idx)
+                (drop cmdline (add1 tout-file-idx))))))
+
+(define (open-cmd-output-port cmd-output-file)
+  (if (path? cmd-output-file)
+      (with-handlers ([exn:fail:filesystem:errno?
+                       (lambda (e)
+                         (displayln e)
+                         (displayln "Failed to open output file, fallback to standard output")
+                         (current-output-port))])
```

Comment:
In general I think it's better to write error to `current-error-port`.

I'm not sure if I like the fallback behavior. Users intentionally include `--cmd-output` because they expect it to output to a file. Letting the program continue running in the fallback mode defies that expectation. Perhaps just letting the exception getting thrown might be better?